internal/provider/resources/resource_team_test.go

```diff
@@ -0,0 +1,237 @@
+package resources_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/iam"
+	astronomerprovider "github.com/astronomer/terraform-provider-astro/internal/provider"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAcc_ResourceTeam(t *testing.T) {
+	namePrefix := utils.GenerateTestResourceName(10)
+
+	workspaceId := os.Getenv("HOSTED_WORKSPACE_ID")
+	deploymentId := os.Getenv("HOSTED_DEPLOYMENT_ID")
+	userId := os.Getenv("HOSTED_USER_ID")
+
+	teamName := fmt.Sprintf("%v_team", namePrefix)
+	resourceVar := fmt.Sprintf("astro_team.%v", teamName)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: astronomerprovider.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { astronomerprovider.TestAccPreCheck(t) },
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccCheckTeamExistence(t, teamName, false),
+		),
+		Steps: []resource.TestStep{
+			// Test failure: disable team resource if org is isScimEnabled
```

Comment:
lets add test where we have deployment but no matching workspace and make sure we get the right err